core/src/test/java/nl/nn/adapterframework/doc/model/FrankDocGroupTest.java

```diff
@@ -0,0 +1,65 @@
+package nl.nn.adapterframework.doc.model;
+
+import java.util.Map;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import nl.nn.adapterframework.doc.ModelBuilder;
+
+public class FrankDocGroupTest {
+	public static final String SIMPLE = "nl.nn.adapterframework.doc.testtarget.simple";
+	private static final String PARENT = SIMPLE + ".ListenerParent";
+	private static final String CHILD = SIMPLE + ".ListenerChild";
+
+	@Test
+	public void whenChildElementAddedBeforeParentThenCorrectModel() {
+		ModelBuilder builder = new ModelBuilder();
+		FrankDocGroup group = builder.getModel().addGroup("Listeners");
+		group.addAsElement(ModelBuilder.getClass(CHILD), builder.getModel());
+		group.addAsElement(ModelBuilder.getClass(PARENT), builder.getModel());
+		checkModelAfterChildAndParentAdded(builder.getModel());
+	}
+
+	@Test
+	public void whenParentElementAddedBeforeChildThenCorrectModel() {
+		ModelBuilder builder = new ModelBuilder();
+		FrankDocGroup group = builder.getModel().addGroup("Listeners");
+		group.addAsElement(ModelBuilder.getClass(PARENT), builder.getModel());
+		group.addAsElement(ModelBuilder.getClass(CHILD), builder.getModel());
+		checkModelAfterChildAndParentAdded(builder.getModel());
+	}
+
+	private void checkModelAfterChildAndParentAdded(FrankDocModel model) {
+		Assert.assertEquals(1, model.getGroups().size());
+		FrankDocGroup actualGroup = model.getGroups().get(0);
+		Map<String, FrankElement> actualAllElements = model.getAllElements();
+		Assert.assertTrue(actualGroup.getElements().containsKey(PARENT));
+		Assert.assertTrue(actualGroup.getElements().containsKey(CHILD));
+		for(String elementInGroup: actualGroup.getElements().keySet()) {
+			FrankElement groupElement = actualGroup.getElements().get(elementInGroup);
+			FrankElement allElement = actualAllElements.get(elementInGroup);
+			Assert.assertSame(
+					"Different objects for group element and all element for element name: " + elementInGroup,
+					groupElement, allElement);
+		}
+		FrankElement actualObject = model.getAllElements().get("java.lang.Object");
+		Assert.assertNull(actualObject.getParent());
+		FrankElement actualParent = actualGroup.getElements().get(PARENT);
+		Assert.assertSame(actualObject, actualParent.getParent());
+		Assert.assertEquals(PARENT, actualParent.getFullName());
```

Comment:
voor assertEquals() en assertSame() kan je static imports maken: 
```
import static org.junit.Assert.assertEquals;
```
Als je in de editor assertEquals\<Ctrl-Space\> typet, komt dat vanzelf.

In de code is dat herkenbaarder, en daardoor beter leesbaar

Reply:
Gedaan.